rdflib/graph.py

```diff
@@ -453,20 +453,23 @@ def remove(self, triple):
         return self
 
     def triples(
-        self, triple: Tuple[Optional[Node], Union[None, Path, Node], Optional[Node]]
-    ):
+        self,
+        triple: Tuple[
+            Optional[IdentifiedNode], Union[None, Path, IdentifiedNode], Optional[Node]
+        ],
+    ) -> Iterable[Tuple[IdentifiedNode, Union[IdentifiedNode, Path], Node]]:
```

Comment:
I think this is more correct than it was, but technically this should be:

```suggestion
        self,
        triple: Tuple[
            Optional[IdentifiedNode], Union[None, Path, URIRef], Optional[Node]
        ],
    ) -> Iterable[Tuple[IdentifiedNode, Union[URIRef, Path], Node]]:
```

Fine with it as is though, we can narrow it more later, this narrow already makes things better.

Same applies other in places also, won't label them all, just a note.